src/spectral.jl

```diff
@@ -179,42 +181,43 @@ A = A + A'
 # using UnicodePlots; lineplot(v)
 ~~~
 """
-function fiedler_vector{V}(A::SparseMatrixCSC{V,Int};
-    tol=1e-12,maxiter=300,dense=96,nev=2,checksym=true)
+function fiedler_vector(A::SparseMatrixCSC{V,Int};
+    tol=1e-12,maxiter=300,dense=96,nev=2,checksym=true) where V
     n = checksquare(A)
     if checksym
         if !issymmetric(A)
             throw(ArgumentError("The input matrix must be symmetric."))
         end
     end
 
-    d = vec(sum(A,1))
-    d = sqrt(d)
+    d = vec(sum(A,dims=1))
+    d = sqrt.(d)
 
     if n == 1
         X = zeros(V,1,2)
         lam2 = 0.
     elseif n <= dense
         ai,aj,av = findnz(A)
         L = sparse(ai,aj,-av./((d[ai].*d[aj])),n,n) # applied sqrt above
-        L = full(L) + 2*eye(n)
-        F = eigfact!(Symmetric(L))
+        L = Matrix(L) + 2I
+        F = eigen!(Symmetric(L))
         lam2 = F.values[2]-1.
         X = F.vectors
     else
         ai,aj,av = findnz(A)
         L = sparse(ai,aj,-av./((d[ai].*d[aj])),n,n) # applied sqrt above
-        L = L + 2.*speye(n)
+        L = L + 2. *sparse(1.0I,n,n)
```

Comment:
Better is L + sparse(2.0I, n,n)

Reply:
This and another occurrence should be fixed here bd18cb2219e2908e02e5af25bbdea86196d615da